lib/qesapdeployment.pm

```diff
@@ -1707,4 +1729,80 @@ sub qesap_export_instances {
     record_info('EXPORT', "SSH keys and instances data uploaded to test results:\n" . join("\n", @upload_files));
 }
 
+=head3 qesap_is_job_finished
+
+    Get whether a specified job is still running or not
+
+=over 1
+
+=item B<JOB_ID> - id of job to check
+
+=back
+=cut
+
+sub qesap_is_job_finished {
+    my ($job_id) = @_;
+    my $url = get_var('OPENQA_HOSTNAME') . "/api/v1/jobs/$job_id";
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;    # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    return ($job_state ne 'running');
+}
+
+
+=head3 qesap_az_get_active_peerings
+
+    Get active peerings for Azure jobs
+
+=over 2
+
+=item B<RG> - Resource group in question
+
+=item B<VNET> - vnet name of rg
+
+=back
+=cut
+
+sub qesap_az_get_active_peerings {
+    my (%args) = @_;
+    foreach (qw(rg vnet)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my $cmd = "az network vnet peering list -g $args{rg} --vnet-name $args{vnet} --output tsv --query \"[].name\"";
+    my $output_str = script_output($cmd);
+    my @output = split(/\n/, $output_str);
+    my %result;
+
+    foreach my $line (@output) {
+        # find integers in the vnet name that are 6 digits or longer - this would be the job id
+        my @matches = $line =~ /(\d{6,})/g;
+        $result{$line} = $matches[-1] if @matches;
+    }
+    return %result;
+}
+
+=head2 qesap_az_clean_old_peerings
+
+    Delete leftover peerings for Azure jobs that finished without cleaning up
+
+=cut
+
+sub qesap_az_clean_old_peerings {
+    my (%args) = @_;
+    foreach (qw(rg vnet)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my %peerings = qesap_az_get_active_peerings(rg => $args{rg}, vnet => $args{vnet});
+
+    while (my ($key, $value) = each %peerings) {
+        if (qesap_is_job_finished($value) == 1) {
```

Comment:
Nice! It worked with `== 1` even when returning an actual bool from the function. For the explanation see:
```
$ perl
$val = 'running' eq 'not running';
print "Result of 'running' eq 'not running' => [$val]\n";
$val = 'running' eq 'running';
print "Result of 'running' eq 'running' => [$val]\n";
Result of 'running' eq 'not running' => []
Result of 'running' eq 'running' => [1]
```
Anyway, `== 1` is not needed, but you can leave it there. I think it's more readable without it.

Reply:
Done!